charts/loki-distributed/templates/query-scheduler/service-query-scheduler.yaml

```
@@ -0,0 +1,30 @@
+{{- if .Values.queryScheduler.enabled }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "loki.querySchedulerFullname" . }}
+  labels:
+    {{- include "loki.querySchedulerLabels" . | nindent 4 }}
+    {{- with .Values.queryScheduler.serviceLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with .Values.loki.serviceAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: http
+      port: 3100
+      targetPort: http
+      protocol: TCP
+    - name: grpclb
```

would it be better to istio friendly?

```suggestion
    - name: lb-grpc
```

---

I tried to stay consistent with [ports name when using Tanka deployment](https://github.com/grafana/loki/blob/main/production/ksonnet/loki/common.libsonnet#L29). So I don't know if we should diverge from it or not, maybe someone has the rationale behind this name?

---

It was also used in the [frontend](https://github.com/grafana/helm-charts/blob/cb6f01bc3bd2ec7ce35880c021942d2c38b6c037/charts/loki-distributed/templates/query-frontend/service-query-frontend.yaml#L27), but ye I don't really know the reason behind the name, think it's related to the fact it resolves to a DNS SRV record?